index.js

```diff
@@ -0,0 +1,56 @@
+const defaults = {
+  routeRules: [
+    '.php',
+    '.aspx',
+    '.ashx',
+    '.ico',
+    '/feed',
+    '/rss',
+    '/null',
+    '/undefined',
+    '/...',
+    '/true',
+    '/false'
+  ],
+  userAgentRules: [
+    'mj12bot' // Generates excessive 404s
+  ]
+};
+
+const register = function(server, options) {
+  const settings = Object.assign({}, defaults, options);
+
+  server.ext('onPreResponse', (request, h) => {
```

Comment:
can we make this an earlier lifecycle point? At this point in the request the entire handler has already been run.

Reply:
So I tried with onPreHandler but it wasn't triggering if the route didn't exist (kinda the whole point). I'll dig in and see if there's something else